### DISCUSSION TITLE
Option to hide deployment history from pull request conversation tab

### DISCUSSION THREAD

**Discussion by vgouv** [community/community#58018] [Actions]
### Select Topic Area

Product Feedback

### Body

When you configure deployment environments with github actions, you can view the deployment history in the actions/Deployments page.
On the other hand this is also present in a pull request's discussion. As this might distract users from following the discussion, there could be a switch (per user configurable) for the deployment to be hidden from the conversation, for users that find this distracting.

**Comment by andrijan** [comment]
Yes please!

We are currently deploying multiple services on every push. So for each push (If the code changes) we deploy to the same environment. When you have multiple pushes the pull request gets cluttered with duplicate events pointing to the same deployment.

**Comment by fokaskostas** [comment]
This would be very useful. The PR's UI is very cluttered especially when there are multiple commits

**Comment by msto** [comment]
Would love to have this feature

**Comment by ttbarnes** [comment]
This would be great - maybe an option to toggle "show/hide all" would suffice.

**Comment by swizes** [comment]
This can't be a problem only to few developer. Scrolling takes more time than reviewing the PR -.-

**Comment by izaak-gynger** [comment]
PLEASE ADD THIS SUPPRESSION FLAG reviews are endless scrolls now and comments get lost.

**Comment by jdiponziano** [comment]
PLEASE add this, it's driving me insane, i hate the PR interface

**Comment by anden-akkio** [comment]
I filter out the deployment events with this uBlock Origin rule:

```
github.com##.js-targetable-element.TimelineItem
```

Tends to occasionally hide a few other things, but the benefit outweighs the downsides in my usage patterns.

**Reply by gmiklich** [reply]
Luckily there is a browser extension (called Refined Github) that does this for you, but even if you don't want to use it, you can see the slightly more accurate selector [here](https://github.com/refined-github/refined-github/blob/main/source/features/hide-inactive-deployments.tsx). Hopefully it helps you avoid hiding the few other things.

**Reply by Cellule** [reply]
I used Refined Github extension, but it hides only inactive deployments. I really wanted to hide them all.
Thankfully the extension allows for custom css so I used the following to hide more, not perfect either, but a good enough in most cases

```css
.js-targetable-element:has(.TimelineItem-badge .octicon-rocket) {
  display: none;
}
```

**Reply by thatkookooguy** [reply]
I hide inactive deployments from timelines in PRs using this css selector with the stylish chrome extension:
```css
.TimelineItem:has(span[title$="Inactive"]) {
  display: none;
}
```

**Comment by OlegMuron** [comment]
Hi!

Please add this possibility - it will be an awesome feature!

**Comment by gmiklich** [comment]
This would be an amazing quality of life improvement when reviewing PRs. The temp deployments provide very little (if any) value to me, and they end up hiding the important conversations that I want to get to.

If github doesn't get around to addressing this, there is a browser extension that allows you to hide such things (among other functionality). I am in no way affiliated with the project, and honestly only found it when searching for a way to hide the messages. [Here's a link ](https://github.com/refined-github/refined-github/tree/main)to the extension (called Refined Github).

**Comment by throttlehead-dev** [comment]
We're using whats got to be a popular pattern where we do build/canary container tags across workflows, where we build and push the canary tag and then use it in a subsequent workflow as a service that can be tested against. This creates a lot of noise on the PRs and has me second guessing switching GitHub to do our production framework builds as well as that repo is **BUSY.**

We should be able to disable the output that gets read in prs in the workflow, similar to how you can modify the url:

```
environment:
	name: prod
	outputToPr: false
```

**Comment by indiarasabarreto** [comment]
https://docs.github.com/pt/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests

**Comment by oleksiial** [comment]
Very needed feature

**Comment by yury-dubinin** [comment]
Is anyone looking at it? I found multiple treads overs the years with the same ask. Example: https://github.com/actions/runner/issues/2120

**Comment by anlandu** [comment]
![image](https://github.com/user-attachments/assets/b0a541d0-5083-438f-8b36-c5e90c2294ac)
🥲

**Reply by Brakjen** [reply]
![image](https://github.com/user-attachments/assets/80e359d8-0c95-4067-9028-915a95a81934)

**Comment by muchisx** [comment]
What's the current workaround for this if it isn't fixed by gh yet?

**Reply by brucewroberts** [reply]
Use gitlab?

**Reply by eirikvaa** [reply]
The Redefined GitHub extension can hide them by using the "Hide events" option, but it comes back after a refresh, so it's not really that useful 😬

**Comment by oskarsword** [comment]
Hello Github really silent on this for whatever reason.
I dont need to see my github actions history in a pull requests thank you.
Maybe make a separate tab for it or whatever.
Can someone handle this as the current design is awful?

**Comment by LeleDallas** [comment]
I actually built a small web plugin to address this!
Feel free to check it out: https://github.com/LeleDallas/Conversation-Cleaner

**Comment by jmank88** [comment]
Who is this information for? It is incredibly noisy, and 99% of developers consider it to be spam.
![Screenshot from 2025-07-07 06-36-16](https://github.com/user-attachments/assets/62148d10-86d5-4e75-bf4d-7bc270a808a7)

**Comment by phsantiago** [comment]
For now, while GitHub doesn't solve it, you can paste this to the console:
```tsx
const style = document.createElement('style');
style.textContent = `
  .js-targetable-element:has(.TimelineItem-badge .octicon-rocket) {
    display: none !important;
  }
`;
document.head.appendChild(style);
```

**Comment by yosefe** [comment]
+1 on this, this creates a huge mess in the PR and takes focus away from real conversations.

**Comment by mbargiel** [comment]
Even with Refined Github, the UX is subpar because if you need to view previous discussions, they could be collapsed under something like 
<img width="229" height="99" alt="image" src="https://github.com/user-attachments/assets/816a6e42-7949-4cf7-b568-791c0d7f4c73" />
which would require you to click a number of times to eventually see them, especially if they're more recent since the "Load more..." button loads _the oldest of those_ first.

**Comment by dkoszewski** [comment]
Please, take this feature from your backlog, prioritize usability and cleanliness over this clutter spam, and do something about it.

**Comment by cino** [comment]
Here to voice my frustration as well, some of the pull-requests we have become literally unworkable because of this.

**Comment by thereis** [comment]
3 years of frustration.

**Comment by mitchdz** [comment]
This is a major pain point for our team. Being able to suppress these messages would make a big difference.

**Comment by phsantiago** [comment]
I'm using a custom comment that gets updated on each deploy:

1. job to post the comment
```
jobs:
  create_preview_comment:
    name: Create preview comment
    if: github.event_name == 'pull_request'
    runs-on: ubuntu-slim
    steps:
      - name: Create or find preview comment
        uses: actions/github-script@v6
        with:
          script: |
            const marker = '<!-- preview-comment -->';
            const prNumber = context.payload.pull_request.number;

            const comments = await github.paginate(github.rest.issues.listComments, {
              ...context.repo,
              issue_number: prNumber,
            });

            if (comments.find(c => c.body?.includes(marker))) {
              core.info('Preview comment already exists');
              return;
            }

            const body = [
              marker,
              '<!--- preview:app --->',
              '**Application Preview:** link will be populated here',
              '<!--- /preview:app --->',
            ].join('\n');

            await github.rest.issues.createComment({
              ...context.repo,
              issue_number: prNumber,
              body,
            });
```

2. step part of the job that makes the deploy, updating the comment
```
    steps:
      - name: Update preview comment
        uses: actions/github-script@v6
        with:
          script: |
            const prNumber = ${{ github.event.pull_request.number }};
            const previewUrl = <your custom preview url here>;
            const sectionRe = /(<!--- preview:app --->)([\s\S]*?)(<!--- \/preview:app --->)/;
            const commentMarker = '<!-- preview-comment -->';

            const comments = await github.paginate(github.rest.issues.listComments, {
              ...context.repo, issue_number: prNumber,
            });
            const comment = comments.find(c => c.body?.includes(commentMarker));
            if (!comment) {
              core.warning('Could not find preview comment');
              return;
            }
            const body = comment.body;

            if (!sectionRe.test(body)) {
              core.warning('Could not find <!--- preview:app ---> section in preview comment');
              return;
            }

            const newBody = body.replace(sectionRe, `$1\n**App Preview:** ${previewUrl}\n$3`);
            if (newBody === body) return;

            await github.rest.issues.updateComment({
              ...context.repo, comment_id: comment.id, body: newBody,
            });
```

**Comment by bdzikuc** [comment]
Hey guys,
I think this is now doable with this [https://github.blog/changelog/2026-03-19-github-actions-late-march-2026-updates/#github-actions-now-allows-developers-to-use-environments-without-auto-deployment](https://github.blog/changelog/2026-03-19-github-actions-late-march-2026-updates/#github-actions-now-allows-developers-to-use-environments-without-auto-deployment)

**Comment by nvm2002** [comment]
good day can i ask whats that?

**Comment by nichita-pasecinic** [comment]
any updates ?

**Comment by Fatiihunza89** [comment]
This seems like a reasonable feature request. Deployment events are valuable for auditing and tracking releases, but they can add significant noise to pull request conversations, especially in repositories with frequent deployments. A user-level setting to collapse or hide deployment history from the Conversation tab would allow developers to focus on code review and discussion while still keeping deployment information available in the Deployments or Actions views when needed.

Making it configurable per user rather than repository-wide would be particularly useful, since some team members may rely on those deployment updates while others find them distracting. It would improve the signal-to-noise ratio without removing any functionality.

**Comment by Cellule** [comment]
On our side, we solved this by taking the following stance.
It is now illegal to use an environment without specifying if it is indeed a deployment or just an environment controlling secrets and other types of access.

## Invalid

```
jobs:
  some-job:
    environment: Tests
```

## Valid

```
jobs:
  some-job:
    environment:
      name: Tests
      deployment: false
``` 

```
jobs:
 actual-deployment:
    environment:
      name: Production
      deployment: true
``` 

Along with a custom validation job that ensures this rule is respected
We now no longer have any deployments appearing in our Pull Requests anymore unless someone actually deploys the branch to a Test environment which is a manual/optional operation.
The fact that using an `environment` creates a "deployment" behind the scene might be a likely default, but in our case we have 94 cases of `deployment: false` and 52 cases of `deployment: true` in our monorepo.
